src/pudl_archiver/archivers/eia/eiacbecs.py

```diff
@@ -0,0 +1,102 @@
+"""Archive EIA  Commercial Buildings Energy Consumption Survey (CBECS)."""
+
+import logging
+import re
+from pathlib import Path
+from urllib.parse import urljoin
+
+from pudl_archiver.archivers.classes import (
+    AbstractDatasetArchiver,
+    ArchiveAwaitable,
+    ResourceInfo,
+)
+from pudl_archiver.frictionless import ZipLayout
+
+BASE_URL = "https://www.eia.gov/consumption/commercial/data/"
+logger = logging.getLogger(f"catalystcoop.{__name__}")
+
+
+class EiaCbecsArchiver(AbstractDatasetArchiver):
+    """EIA CBECS archiver."""
+
+    name = "eiacbecs"
+
+    async def get_resources(self) -> ArchiveAwaitable:
+        """Download EIA-CBECS resources."""
+        link_pattern = re.compile(r"commercial/data/(\d{4})/$", re.IGNORECASE)
+
+        for link in await self.get_hyperlinks(BASE_URL, link_pattern):
+            match = link_pattern.search(link)
+            year = match.group(1)
+            yield self.get_year_resources(year)
+
+    async def get_year_resources(self, year: int) -> list[ResourceInfo]:
+        """Download all files from all views for a year."""
+        data_paths_in_archive = set()
+        zip_path = self.download_directory / f"eiacbecs-{year}.zip"
+        pattern = rf"(?:{year}|archive)(?:.*)/([a-z,\d]{{1,8}})(.xls|.xlsx|.pdf)$"
+        data_view_patterns = {
+            "characteristics": re.compile(pattern),
+            "consumption": re.compile(pattern),
+            "mircodata": re.compile(
+                rf"(?:{year}/|archive/|)(?:xls|pdf|csv)/(.*)(.xls|.xlsx|.pdf|.csv)$"
+            ),
```

Comment:
I had to add this `|)` or nothing bc 1999 data has links like this `csv/file01.csv`

i think i should make a pattern exception for 1999 for this because w/o it we are getting some of the 2018 pdfs on the right column the page.

Reply:
fixed!